simapp/test_simapp.go

```diff
@@ -0,0 +1,358 @@
+package simapp
+
+import (
+	"encoding/json"
+	"math/rand"
+	"reflect"
+	"time"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmjson "github.com/tendermint/tendermint/libs/json"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	bam "github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/client"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/simapp/helpers"
+	"github.com/cosmos/cosmos-sdk/testutil/mock"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	// cheqd specific imports
+	cheqdapp "github.com/cheqd/cheqd-node/app"
+	didtypes "github.com/cheqd/cheqd-node/x/did/types"
+	resourcetypes "github.com/cheqd/cheqd-node/x/resource/types"
+)
+
+// NewSimappWithCustomOptions initializes a new SimApp with custom options.
+func NewSimappWithCustomOptionsForGinkgo(isCheckTx bool, options SetupOptions) *SimApp {
+	privVal := mock.NewPV()
+	pubKey, err := privVal.GetPubKey()
+	if err != nil {
```

Comment:
Those panic exits will probably need to be replaced to catch errors gracefully in Ginkgo.
You can wrap them in a function that runs in a `BeforeSuite` node.

Same goes for all panic exits onwards.

Reply:
```
var _ = BeforeSuite(func() {
	defer func() {
		if err := recover(); err != nil {
			Expect(err).To(BeNil(), "panic occured")
		}
	}()
})
```

@Eengineer1 I defined the BeforeSuite node like this. Now how do I wrap them in every function?